subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/junit5/JUnitPlatformTestClassProcessor.java

```diff
@@ -0,0 +1,236 @@
+package org.gradle.api.internal.tasks.testing.junit5;
+
+import org.gradle.api.internal.tasks.testing.*;
+import org.gradle.api.tasks.testing.TestOutputEvent;
+import org.gradle.api.tasks.testing.TestResult;
+import org.gradle.internal.time.Clock;
+import org.junit.platform.engine.TestExecutionResult;
+import org.junit.platform.engine.TestSource;
+import org.junit.platform.engine.reporting.ReportEntry;
+import org.junit.platform.engine.support.descriptor.ClassSource;
+import org.junit.platform.engine.support.descriptor.MethodSource;
+import org.junit.platform.launcher.*;
+import org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder;
+import org.junit.platform.launcher.core.LauncherFactory;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static org.junit.platform.engine.discovery.DiscoverySelectors.selectClass;
+
+public class JUnitPlatformTestClassProcessor implements TestClassProcessor {
+    private final Clock clock;
+    private Launcher launcher;
+
+    public JUnitPlatformTestClassProcessor(Clock clock) {
+        this.clock = clock;
+    }
+
+    @Override
+    public void startProcessing(TestResultProcessor resultProcessor) {
+        this.launcher = LauncherFactory.create();
+        launcher.registerTestExecutionListeners(new GradleTestExecutionListener(new InspectingTestResultProcessor(resultProcessor), clock));
```

Comment:
Agreed. The issue is that I can't see where the events are being dropped. I can't recall the specifics, since it's been a couple weeks but I spent a good amount of time in the debugger trying to trace when the events where going through and when they weren't. As far as I can tell my listener is getting fired when it should and passing things to Gradle's result processor. Somewhere after that things are getting lost.

Clearly those processors work, so it must be in how I'm using/configuring them. If anyone on your side has thoughts on where that's going wrong, it would be much appreciated.

Reply:
Our team will try to have a look at it soon.